src/town_cmd.cpp

```diff
@@ -1087,6 +1087,59 @@ static bool GrowTownWithRoad(const Town *t, TileIndex tile, RoadBits rcmd)
 	return false;
 }
 
+/**
+ * Checks if a town road can be continued on the other side of a bridge.
+ *
+ * @param end_tile The end tile of the bridge
+ * @param bridge_dir The direction of the bridge
+ * @return true if the road can be continued, else false
+ */
+static bool CanContinueRoad(const TileIndex end_tile, const DiagDirection bridge_dir)
+{
+	const int delta = TileOffsByDiagDir(bridge_dir); // +1 tile in the direction of the bridge
+	TileIndex next_tile = end_tile + delta; // The tile beyond the bridge
+
+	/* Before we try anything, make sure the tile is on the map and not the void. */
+	if (!IsValidTile(end_tile + delta)) return false;
+
+	/* If the tile is clear or has trees, it is fine to build. */
+	if (IsTileType(next_tile, MP_CLEAR) || IsTileType(next_tile, MP_TREES)) return true;
```

Comment:
This is a lie! Sloping can prevent building :) Here is where the `IsRoadAllowedHere` comes into play. I think you can change this if, and the next 2, with only a call to this function. I might be wrong :)

I would do:

- TunnelBridge check
- Station check
- `IsRoadAllowedHere` call

Reply:
`IsRoadAllowedHere` didn't work, but I used a `DoCommand` like your originally suggested for the same purpose.